src/block-management/block-holder.js

```diff
@@ -45,36 +62,109 @@ export class BlockHolder extends React.Component<PropsType, StateType> {
+				TextInputState.blurTextInput( currentlyFocusedTextInput );
+			}
 		}
+		this.props.onSelect( this.props.clientId );
+	};
 
-		// Return empty view, toolbar won't be rendered
-		return <View />;
+	onInlineToolbarButtonPressed = ( button: number ) => {
+		switch ( button ) {
+			case InlineToolbarActions.UP:
+				this.props.moveBlockUp();
+				break;
+			case InlineToolbarActions.DOWN:
+				this.props.moveBlockDown();
+				break;
+			case InlineToolbarActions.DELETE:
+				this.props.removeBlock();
+				break;
+		}
+	};
+
+	insertBlocksAfter = ( blocks: Array<Object> ) => {
+		const order = this.props.getBlockIndex( this.props.clientId, this.props.rootClientId );
+		this.props.onInsertBlocks( blocks, order + 1 );
+
+		if ( blocks[ 0 ] ) {
+			// focus on the first block inserted
+			this.props.onSelect( blocks[ 0 ].clientId );
+		}
+	};
+
+	mergeBlocks = ( forward: boolean = false ) => {
+		const {
+			clientId,
+			getPreviousBlockClientId,
+			getNextBlockClientId,
+			mergeBlocks,
+		} = this.props;
+
+		const previousBlockClientId = getPreviousBlockClientId( clientId );
+		const nextBlockClientId = getNextBlockClientId( clientId );
+
+		// Do nothing when it's the first block.
+		if (
+			( ! forward && ! previousBlockClientId ) ||
+			( forward && ! nextBlockClientId )
+		) {
+			return;
+		}
+
+		if ( forward ) {
+			mergeBlocks( clientId, nextBlockClientId );
+		} else {
+			mergeBlocks( previousBlockClientId, clientId );
+		}
+	};
+
+	onBlockHolderLayout = ( event: LayoutChangeEvent ) => {
+		const { width: fullWidth } = Dimensions.get( 'window' );
+		const { width } = event.nativeEvent.layout;
+		const isFullyBordered = fullWidth > width;
+		if ( isFullyBordered !== this.state.isFullyBordered ) {
+			this.setState( { ...this.state, isFullyBordered } );
+		}
+	}
+
+	blockHolderFocusedStyle() {
+		return this.state.isFullyBordered ? styles.blockHolderFocusedFullBordered : styles.blockHolderFocusedSemiBordered;
+	}
+
+	renderToolbar() {
+		if ( ! this.props.isSelected ) {
+			return null;
+		}
+
+		return (
+			<InlineToolbar
+				clientId={ this.props.clientId }
+				onButtonPressed={ this.onInlineToolbarButtonPressed }
+				canMoveUp={ ! this.props.isFirstBlock }
+				canMoveDown={ ! this.props.isLastBlock }
+			/>
+		);
 	}
 
 	getBlockForType() {
 		return (
 			<BlockEdit
 				name={ this.props.name }
 				isSelected={ this.props.isSelected }
-				attributes={ { ...this.props.attributes } }
-				// pass a curried version of onChanged with just one argument
-				setAttributes={ ( attrs ) =>
-					this.props.onChange( this.props.clientId, { ...this.props.attributes, ...attrs } )
-				}
-				onFocus={ this.props.onSelect }
-				onReplace={ this.props.onReplace }
-				insertBlocksAfter={ this.props.insertBlocksAfter }
-				mergeBlocks={ this.props.mergeBlocks }
+				attributes={ this.props.attributes }
+				setAttributes={ this.props.onChange }
+				onFocus={ this.onFocus }
+				onReplace={ this.props.replaceBlock }
```

Comment:
Hmm, it seems `replaceBlock()` has a very different signature than `onReplace()` that was passed before. Are we sure `BlockEdit` is OK with the `replaceBlock()`?

Reply:
Indeed, I must have messed up here, fixing that 👍